pallets/statistics/src/lib.rs

```diff
@@ -54,273 +74,813 @@ decl_module! {
+        fn on_runtime_upgrade() -> Weight {
+            storage_migrate_on!(StorageVersion::get(), 1, {
+                migration::migrate_v1::<T>();
+            });
 
-        /// Adds a new transfer manager.
-        ///
-        /// # Arguments
-        /// * `origin` It contains the secondary key of the caller (i.e who signed the transaction to execute this function).
-        /// * `ticker` ticker for which the transfer managers are being updated.
-        /// * `new_transfer_manager` Transfer manager being added.
-        ///
-        /// # Errors
-        /// * `Unauthorized` if `origin` is not the owner of the ticker.
-        /// * `DuplicateTransferManager` if `new_transfer_manager` is already enabled for the ticker.
-        /// * `TransferManagersLimitReached` if the `ticker` already has max TMs attached
+            0
+        }
+
+        const MaxStatsPerAsset: u32 = T::MaxStatsPerAsset::get();
+        const MaxTransferConditionsPerAsset: u32 = T::MaxTransferConditionsPerAsset::get();
+
+        /// Set the active asset stat_types.
         ///
-        /// # Permissions
+        /// # Permissions (EA)
         /// * Asset
-        #[weight = <T as Config>::WeightInfo::add_transfer_manager()]
-        pub fn add_transfer_manager(origin, ticker: Ticker, new_transfer_manager: TransferManager) {
-            let did = <ExternalAgents<T>>::ensure_perms(origin, ticker)?;
-            ActiveTransferManagers::try_mutate(&ticker, |transfer_managers| {
-                ensure!((transfer_managers.len() as u32) < T::MaxTransferManagersPerAsset::get(), Error::<T>::TransferManagersLimitReached);
-                ensure!(!transfer_managers.contains(&new_transfer_manager), Error::<T>::DuplicateTransferManager);
-                transfer_managers.push(new_transfer_manager.clone());
-                Ok(()) as DispatchResult
-            })?;
-            Self::deposit_event(Event::TransferManagerAdded(did, ticker, new_transfer_manager));
+        #[weight = <T as Config>::WeightInfo::set_active_asset_stats(stat_types.len() as u32)]
+        pub fn set_active_asset_stats(origin, asset: AssetScope, stat_types: BTreeSet<StatType>) {
+            Self::base_set_active_asset_stats(origin, asset, stat_types)?;
         }
 
-        /// Removes a transfer manager.
-        ///
-        /// # Arguments
-        /// * `origin` It contains the secondary key of the caller (i.e who signed the transaction to execute this function).
-        /// * `ticker` ticker for which the transfer managers are being updated.
-        /// * `transfer_manager` Transfer manager being removed.
-        ///
-        /// # Errors
-        /// * `Unauthorized` if `origin` is not the owner of the ticker.
-        /// * `TransferManagerMissing` if `asset_compliance` contains multiple entries with the same `requirement_id`.
+        /// Allow a trusted issuer to init/resync ticker/company stats.
         ///
-        /// # Permissions
+        /// # Permissions (EA)
```

Comment:
`# Permissions` is required for the permission script. Maybe we can standardize the format for EA vs non-EA perms to solve this. However, in its current form this won't get picked up by the script.

Reply:
Fixed.  Thanks.